front/src/service.ts

```diff
@@ -420,4 +421,71 @@ export class ScratchyService {
         this.messageCache.delete(messageID);
         return response.data;
     }
-}
\ No newline at end of file
+
+/* Call server with Websocket */
+
+     /**
+     *
+     * @async
+     * @augments ScratchyService
+     * @param {Room} room - affected room
+     * @returns {Promise<undefined>} -
+     */
+    async joinRoomWebsocket(room: Room): Promise<void> {
+        socket.on("join", (socket) => {  
+            socket.join(room.id);
+        });
+    }
+
+     /**
+     *
+     * @async
+     * @augments ScratchyService
+     * @param {Room} room - affected room
+     * @returns {Promise<undefined>} -
+     */
+      async quitRoomWebsocket(room: Room): Promise<void> {
+        socket.on("quit", (socket) => {  
+            socket.leave(room.id);
+        });
+    }
+
+    /**
+     *
+     * @async
+     * @augments ScratchyService
+     * @returns {Promise<Message>} - message data
+     */
+     async newMessageWebsocket(): Promise<void> {
+        socket.on("newMessage", (data) => {
+            return data;
+        });      
+     }   
+
+    /**
+     * 
+     *
+     * @async
+     * @param {string} userId - id of the user
+     * @param {Room} room - affected room
+     * @augments ScratchyService
+     * @returns {Promise} - message data
+     */
+     async start_writing(userId : string ,room: Room): Promise<void> {
+                socket.emit("writingState",{userId : userId, status: "start_writing" });     
```

Comment:
In viandox's chart there is 2 different event:
- start_writing
- stop_writing
not a particular info status in the raquest

Reply:
done here [7b92067](https://github.com/Draeggon/scratchy/pull/2/commits/7b92067314a445581480d40413c943e4dbf6f85e)